File: Servidor-Java-main/src/main/java/com/jacs/proyectomicroservicios/dto/MovimientoConTitularDTO.java
package com.jacs.proyectomicroservicios.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * DTO — Consulta personalizada 1 (TERCER PROTOTIPO).
 *
 * "El listar de la Clase/Tabla B debe mostrar todos sus atributos más la
 *  llave foránea y un atributo de la Clase/Tabla A."
 *
 * Retorna todos los campos de MOVIMIENTO (Tabla B) más {@code titularCuenta}
 * obtenido de CUENTA_AHORROS (Tabla A) mediante JOIN JPQL.
 *
 * Endpoint: GET /cuentas/movimientos/filtrar
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class MovimientoConTitularDTO {

    /** ID del movimiento (Tabla B — PK). */
    private Integer id;

    /** Fecha y hora del movimiento. */
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime fechaMovimiento;

    /** Monto de la transacción. */
    private double monto;

    /** Tipo: "CREDITO" o "DEBITO". */
    private String tipo;

    /** FK hacia CUENTA_AHORROS (llave foránea explícita). */
    private int numeroCuenta;

    /** Titular de la cuenta — atributo de Tabla A (CUENTA_AHORROS). */
    private String titularCuenta;
}

File: Cliente-WinForms/model/MovimientoConTitularDto.cs
using System;
using Newtonsoft.Json;

namespace WayBankClient.model
{
    /// <summary>
    /// DTO — Consulta personalizada #1 (Tercer Prototipo).
    /// Recibe los datos de GET /movimientos/filtrar en MS-Movimiento (8081).
    /// Incluye todos los atributos de MOVIMIENTO + titular de CUENTA_AHORROS.
    /// </summary>
    public class MovimientoConTitularDto
    {
        public int    Id           { get; set; }
        public double Monto        { get; set; }
        public string Tipo         { get; set; }
        public int    NumeroCuenta { get; set; }

        /// <summary>Titular de la cuenta — atributo de CUENTA_AHORROS (Tabla A).</summary>
        public string TitularCuenta { get; set; }

        [JsonProperty("fechaMovimiento")]
        public string FechaMovimientoStr { get; set; }

        public DateTime FechaMovimiento
        {
            get
            {
                if (string.IsNullOrEmpty(FechaMovimientoStr)) return DateTime.MinValue;
                return DateTime.Parse(FechaMovimientoStr);
            }
        }

        public override string ToString()
        {
            return $"[{Id}] {TitularCuenta} — {Tipo} ${Monto:N0} ({NumeroCuenta})";
        }
    }
}
